File: solutions/java/hamming/1/src/main/java/Hamming.java
public class Hamming {
    String leftDNAStrand;
    String rightDNAStrand;

    public Hamming(String leftStrand, String rightStrand) {
        leftDNAStrand = leftStrand;
        rightDNAStrand = rightStrand;
        // Checking lengths:
        int leftDNALength = leftDNAStrand.length();
        int rightDNALength = rightDNAStrand.length();
        if (leftDNALength != rightDNALength) {
            if (leftDNALength == 0) { throw new IllegalArgumentException("left strand must not be empty."); }
            if (rightDNALength == 0) { throw new IllegalArgumentException("right strand must not be empty."); }
            throw new IllegalArgumentException("leftStrand and rightStrand must be of equal length.");
        }
    }

    public int getHammingDistance() {
        int hammingDistance = 0;
        for(int index = 0; index < leftDNAStrand.length(); index++) {
            if (leftDNAStrand.charAt(index) != rightDNAStrand.charAt(index)) {
                hammingDistance += 1;
            }
        }
        return hammingDistance;
    }
}

File: solutions/javascript/hamming/2/hamming.js
//
// This is only a SKELETON file for the 'Hamming' exercise. It's been provided as a
// convenience to get you started writing code faster.
//

export const compute = (leftStrand, rightStrand) => {
  verifyLength(leftStrand, rightStrand)
  return [...leftStrand].filter((cell, index) => cell !== rightStrand[index]).length;
};

const verifyLength = (leftStrand, rightStrand) => {
  if (leftStrand.length !== rightStrand.length) {
    if (!leftStrand) { throw new Error("left strand must not be empty"); }
    if (!rightStrand) { throw new Error("right strand must not be empty"); }
    throw new Error("left and right strands must be of equal length");
  }
}
